.github/workflows/tests-mac-x86.yml

```diff
@@ -0,0 +1,54 @@
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  build-supergraph:
+    name: Build Supergraph for macOS x86-64
+    # Arm64 runner for cross-compilation
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+        name: "Checkout federation-rs repo"
+      - uses: actions-rs/toolchain@v1
+        name: "Install x86_64 toolchain for cross-compilation"
+        with:
+          toolchain: stable
+          target: x86_64-apple-darwin
+      - run: |
+          rustup target add x86_64-apple-darwin
+      # Build supergraph
+      - uses: actions-rs/cargo@v1
+        name: "Build supergraph binary"
+        with:
+          command: build
+          args: --target x86_64-apple-darwin
+      - uses: actions/upload-artifact@v4
+        name: "Store built binaries to use later on"
+        with:
+          path: |
+            target/x86_64-apple-darwin/debug/supergraph
+          if-no-files-found: error
+          retention-days: 5
+  comopse-supergraph:
+    needs: build-supergraph
+    name: Run supergraph compose on macOS x86-64
+    # strategy:
+    #   matrix:
+    #     composition-version: ${{ fromJSON(inputs.composition-versions) }}
+    # x86-64 runner
+    runs-on: macos-13
+    env:
+      TEST_COMPOSITION_VERSION: "=${{ matrix.composition-version }}"
```

Comment:
These env. vars. are not set after all. And, it works fine without them.
Do we still need these lines (and the commented-out `strategy` section up there)?